core/src/main/java/org/wso2/msf4j/internal/MicroservicesRegistryImpl.java

```diff
@@ -92,8 +92,13 @@ public void addService(String basePath, Object service) {
     }
 
     public void removeService(Object service) {
-        services.remove(service);
-        updateMetadata();
+        Path path = service.getClass().getAnnotation(Path.class);
+        if (path != null) {
+            services.remove(path.value());
+            updateMetadata();
+        } else {
+            log.warn("Service removal fail, class '" + service.getClass().getName() + "' doesn't contain a root Path");
```

Comment:
Shall we change this log to `Service removal failed. Microservice class '" + service.getClass().getName() + "' doesn't contain a root Path.`